src/groups/mqb/mqba/mqba_application.cpp

```diff
@@ -259,6 +262,13 @@ int Application::start(bsl::ostream& errorDescription)
         }
     }
 
+    mqbi::ClusterResources resources;
+
+    resources.d_blobSpPool_p       = &d_blobSpPool;
+    resources.d_bufferFactory_p    = &d_bufferFactory;
+    resources.d_scheduler_p        = d_scheduler_p;
+    resources.d_pushElementsPool_p = &d_pushElementsPool;
```

Comment:
What about passing these fields to the constructor instead? My concern is about adding a new field in resources, with a field access like here it's easy to forget a new field somewhere. With a constructor extended with a new field, the compiler will point to all places where a new field is missing

Reply:
Yeah, that was the old way and the constructor signature kept growing.  I thought, it would help to encapsulate everything in `Resources` so when we need something new from `mqba::Application` far away from it, we don't have to change signatures across the entire chain of constructors.
The concern about missing field is addressed by the fact that if a component needs the field, the compilation will fail.
The concern about missing field value can be addressed by an assert